### PR TITLE
nixos/vaultwarden: backup all rsa_keys

### PR DESCRIPTION
**Pull request by caffineehacker**
The official documentation mentions rsa_key* as what should be backed up (https://github.com/dani-garcia/vaultwarden/wiki/Backing-up-your-vault#the-rsa_key-files). My particular install has rsa_key.pem and rsa_key.pub.pem so the existing command fails when trying to copy rsa_key.der. This change better aligns with the official documentation.

## Description of changes

This is a simple change to just copy all rsa_key.* files instead of specific extensions. This commit to vaultwarden actually removed the use of .der files: https://github.com/dani-garcia/vaultwarden/commit/46e0f3c43a81ce9411612c152e414162a9c220ac

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by caffineehacker** [comment]
@dotlambda @SuperSandro2000 looks like this didn't get auto assigned

**Comment by SuperSandro2000** [comment]
> My particular install has rsa_key.pem and rsa_key.pub.pem so the existing command fails when trying to copy rsa_key.der. This change better aligns with the official documentation.

same for mine

**Review by SuperSandro2000 — APPROVED** [review]
LGTM

**Comment by dotlambda** [comment]
Why don't we backup `sends` and `config.json`?

**Comment by caffineehacker** [comment]
> Why don't we backup `sends` and `config.json`?

We probably should, but note that config.json is only there if the user has changed settings via the admin page (I think, may get created in other ways, but it won't be in every install). We should probably do a command that copies everything in data except the sqlite files. I can take a look at making that work tomorrow.

File: nixos/modules/services/security/vaultwarden/backup.sh
Comment (SuperSandro2000) [inline nixos/modules/services/security/vaultwarden/backup.sh:17]:
```suggestion
if [[ -e "$DATA_FOLDER"/config.json ]]; then
  cp -r "$DATA_FOLDER"/config.json "$BACKUP_FOLDER"
fi
cp -r "$DATA_FOLDER"/icon_cache "$BACKUP_FOLDER"
cp -r "$DATA_FOLDER"/sends "$BACKUP_FOLDER"
```

**Comment by SuperSandro2000** [comment]
Lets merge #292857 first, to know if the backups work at all.

File: nixos/modules/services/security/vaultwarden/backup.sh
Comment (caffineehacker) [inline nixos/modules/services/security/vaultwarden/backup.sh:17]: @SuperSandro2000 What are your thoughts on doing the same logic before doing the sqlite dump? Technically this backup script could be used with another backing DB (although a lot less useful). I know the script still continues executing, but it does produce error messages in the logs.

Comment (SuperSandro2000) [reply nixos/modules/services/security/vaultwarden/backup.sh:17]: sure, why not

**Comment by SuperSandro2000** [comment]
Can you rebase to draw in the above PR and adjust the test accordingly? That would be awesome

**Comment by caffineehacker** [comment]
> Can you rebase to draw in the above PR and adjust the test accordingly? That would be awesome

Done. I've done some further refinements which should reduce unnecessary errors. I'm on the fence as to whether not having the script return code 0 if the data directory doesn't exist is the right thing or not. It doesn't fail the tests to not have it, but it does show up in red during testing and in the journald logs.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by caffineehacker** [comment]
Please let me know if there is anything else waiting on me before this can be merged. No rush, just want to make sure I didn't miss an action item.

**Pull request merged**
